revoke/revoke_test.go

```diff
@@ -129,35 +166,68 @@ func mustParse(pemData string) *x509.Certificate {
 	return cert
 }
 
+func insertLocalCRL(revoke *Revoke) {
+	cert, err := x509.ParseCRL([]byte(localCRLList))
+	if err != nil {
+		panic(err.Error())
+	}
+	revoke.localCRL = localCRLPath
+	revoke.crlSet[localCRLPath] = cert
+}
+
 func TestRevoked(t *testing.T) {
-	if revoked, ok := VerifyCertificate(revokedCert); !ok {
+	rc := New(false)
+	if revoked, ok := rc.VerifyCertificate(revokedCert); !ok {
+		fmt.Fprintf(os.Stderr, "Warning: soft fail checking revocation")
+	} else if !revoked {
+		t.Fatalf("revoked certificate should have been marked as revoked")
+	}
+
+	insertLocalCRL(rc)
```

Comment:
I see insertLocalCRL makes no difference. Also, this insertLocalCRL is a short cut. It hardly tests any file fetch logic you put in.

And we can now test directly with functions defined directly (and test defaultChecker's thread safety). The reason I suggested using new checker at every test is that I saw you spent too much time writing the locking logic
at that time is not yet proper. With a fresh checker with every test, we can focus on code logic first.